Calculator/Form1.cs

```diff
@@ -295,6 +295,133 @@ private void PerformOperation()
                         textbox_result.Text = resultValue.ToString();
                         break;
                     }
+                case Operation.RaiseNumberToPower:
+                    {
+                        resultValue = Math.Pow(firstTerm, secondTerm);
+                        textbox_result.Text = resultValue.ToString();
+                        break;
+                    }
+                case Operation.SquareRoot:
+                    {
+                        resultValue = Math.Sqrt(firstTerm);
+                        textbox_result.Text = resultValue.ToString();
+                        break;
+                    }
+                case Operation.NumberParity:
+                    {
+                        if (firstTerm % 2 == 0)
+                        {
+                            lbl_title.Text = "The number is even.";
+                        }
+                        else
+                        {
+                            lbl_title.Text = "The number is odd.";
+                        }
+                        break;
+                    }
+                case Operation.MirroredNumber:
+                    {
+                        uint resultValue = 0;
+                        uint number = Convert.ToUInt32(firstTerm);
+                        while (number != 0)
+                        {
+                            resultValue = resultValue * 10 + number % 10;
+                            number /= 10;
+                        }
+                        lbl_title.Text = $"The mirrored number is {resultValue}";
+                        break;
+                    }
+                case Operation.IsPrime:
+                    {
+                        uint number = Convert.ToUInt32(firstTerm);
+                        if (number == 0 || number == 1)
+                        {
+                            lbl_title.Text = $"{number} is not a prime number.";
+                        }
+                        for (uint divisor = 2; divisor <= Math.Sqrt(firstTerm); divisor++)
+                        {
+                            if (number % divisor == 0)
+                            {
+                                lbl_title.Text = $"{number} is not a prime number.";
+                            }
+                            else
+                            {
+                                lbl_title.Text = $"{number} is  a prime number.";
+                            }
+                        }
+                        break;
+                    }
+                case Operation.PalindromeSuperPalindrome:
+                    {
+                        uint number = Convert.ToUInt32(firstTerm);
```

Comment:
There is no point to keep converting to uint here if you're not doing any validation

Reply:
Went to mirrored method and converted the uint result to double then deleted the conversions i had before in the PalindromeSuperPalindrome method

Reply:
to GetMirrored*